src/main/kotlin/app/hashers/RepoHasher.kt

```diff
@@ -38,8 +47,16 @@ class RepoHasher(private val localRepo: LocalRepo, private val api: Api,
             getRepoFromServer()
 
             // Hash by all plugins.
-            CommitHasher(localRepo, repo, api, git).update()
-            CodeLongevity(localRepo, repo, api, git).update()
+            val observable = CommitObservable.getObservable(git, serverRepo)
+                                             .publish()
+            CommitHasher(localRepo, serverRepo, api, rehashes)
+                .updateFromObservable(observable)
```

Comment:
I would probably rename the method to chainWith(observable) or JSish then(observable) or you could wrap that by some fancy interfaces like (names and code are sketchy)
interface Thenable {
  then(commits: Observable);
};
FunHasher : Thenable {
  override then(Observable)
}
and then
Commits.asObservalbe().
then(FunHasher).
then(CommitHasher)